### DISCUSSION TITLE
Problem with PHP 8.1 / Error 503 / all-inkl.com

### DISCUSSION THREAD

**Discussion by StefanGebhardt** [kimai/kimai#3717] [On-Premise: Installation & Updates]
Hi,

i have a perfect running installation on all-inkl.com with PHP 8.0. Installed as an older version with PHP 7.4 (or 7.3) and then upgraded without problems.

When I tried to upgrade to PHP 8.1 I got after some seconds an "503 Service Unavailable". Downgrade to PHP 8.0 and it worked again perfect.

So I did a fresh testinstallation step by step from the actual documentation in another path with 8.1 - same problem.

SSH-Upgrades are working fine on both installations (today to 1.30.1) - despite of these 3 problems:
```
Package symfony/debug is abandoned, you should avoid using it. Use symfony/error-handler instead.
Package symfony/inflector is abandoned, you should avoid using it. Use EnglishInflector from the String component instead.
Package php-cs-fixer/diff is abandoned, you should avoid using it. No replacement was suggested.

```
Logfile is empty.

Any hints, where I can start my search (found one similar discussion here https://github.com/kimai/kimai/discussions/2859, but that dos not really help)? 
Are the errors a problem? And if, how can I fix them?

Thanks,
Stefan

**Comment by kevinpapst** [comment]
https://github.com/kimai/kimai/issues/2868
https://github.com/kimai/kimai/discussions/2859
https://github.com/symfony/symfony/discussions/44270#discussioncomment-3445815

In short: stay on 8.0 or leave all-inkl.com - they do not care about fixing the problem.

**Reply by StefanGebhardt** [reply]
Thank you, @kevinpapst

**Reply by StefanGebhardt** [reply]
Hi, I had a support call with all-inkl and they could help with a workaround. In the menatime I upgraded my test stage to 2.0.0-alpha and it works with PHP 8.1 and this workaround.

But I can not evaluate their proposal in general...

"Wir gehen davon aus das das Problem hier eher beim Doctrine-Cache zu suchen ist, welche nicht mehr weiter entwickelt wird. Sie sollten hier auf eine Alternative ausweichen.

Als Workaround habe ich nun in der Datei "./config/packages/cache.yaml" die Zeile 18 deaktiviert und dafür die Zeile 19 hinzugefügt."

```
            doctrine.system_cache_pool:
#                adapter: cache.system       (line 18)
                adapter: cache.adapter.filesystem      (line 19)
```

**Reply by kevinpapst** [reply]
> Wir gehen davon aus das das Problem hier eher beim Doctrine-Cache zu suchen ist, welche nicht mehr weiter entwickelt wird. Sie sollten hier auf eine Alternative ausweichen.

Nie um eine Ausrede verlegen... wenn ioncubeLoader Dateien interpretiert die ihn nichts angehen, ist das ein Produktfehler vom ioncubeLoader. Der Workaround sollte nicht notwendig sein mit 2.0-alpha, bitte testen !

**Reply by StefanGebhardt** [reply]
Habe ich mit 2.0.0-alpha heute getestet. Ohne workaround der 503, mit workaround geht es

**Reply by kevinpapst** [reply]
Wird ja immer merkwürdiger. Der Cache `cache.system` sollte sowieso nur ein Alias sein für `cache.adapter.filesystem`.
Danke fürs Testen, ist ja wunderbar das es doch einen Workaround gibt.
